controller/composite/controller.go

```diff
@@ -232,17 +233,25 @@ func (pc *parentController) enqueueParentObject(obj interface{}) {
 }
 
 func (pc *parentController) updateParentObject(old, cur interface{}) {
-	// Ignore updates where the ResourceVersion changed (not a resync)
-	// but the spec hasn't changed (e.g. our own status updates).
-	// TODO(metacontroller#26): Use ObservedGeneration instead of this.
+	// Ignore updates if the spec hasn't changed (e.g. our own status updates).
 	oldParent := old.(*unstructured.Unstructured)
 	curParent := cur.(*unstructured.Unstructured)
-	if curParent.GetResourceVersion() != oldParent.GetResourceVersion() {
+	oldStatus := k8s.GetNestedField(oldParent.UnstructuredContent(), "status")
+
+	// Status subresource is not found, so it means the subresource is not supported.
```

Comment:
Yes, this part has to be replaced with `ResourceMap` check. I didn't add a TODO here because this PR is already marked as a WIP. We should add TODOs if we want this PR to merge without incorporating changes covered by the TODOs.